simnode/generator.py

```diff
@@ -85,50 +87,166 @@ def _calculate_parallel_coords(offset: float, line_width: float) \
+                                        if offset[i] > 0:
+                                            line_str_0.reverse()
+                                            line_str_1.reverse()
+                                        offset_sub_lines[0].extend(line_str_0)
+                                        offset_sub_lines[1].extend(line_str_1)
+                                    except Exception:
+                                        # this break fixes failing calculation, if a line segment is empty, but
+                                        # it is really ugly
+                                        break
+                            except (NotImplementedError, Exception):  # FIXME Where is TopologyException
+                                _logger.exception("Creating an offset line for lane markings failed")
+                                return None
+                        previous_p = p
+                        i += 1
+                    if offset_sub_lines.__len__() > 1:
+                        # softens stair stepping if the road width changes fast
+                        # TODO tolerance parameter may need to be adjusted
+                        # could be expensive --> replace with #coords = offset_sub_lines
+                        point_list = list(map(lambda i: (offset_sub_lines[0][i], offset_sub_lines[1][i]),
+                                              range(0, offset_sub_lines[0].__len__()-1)))
+                        lstr = LineString(point_list)
+                        lstr = lstr.simplify(tolerance=0.13)
+                        coords = lstr.coords.xy
+
+                        # for plotting the generated road, useful for debugging
+                        '''import matplotlib.pyplot as plt
+                        plt.plot(offset_sub_lines[0], offset_sub_lines[1])
+                        plt.ylabel("first offset: " + str(offset[0]))
+                        plt.show()
+                        '''
+                    # NOTE The parallel LineString may have a different number of points than initially given
+                    num_coords = len(coords[0])
+                    z_vals = repeat(0.01, num_coords)
+                    marking_widths = repeat(line_width, num_coords)
+                    return list(zip(coords[0], coords[1], z_vals, marking_widths))
+
                 # Draw side lines
                 num_lines = road.left_lanes + road.right_lanes + 1
                 initial_line_offsets = [d - (road_width / 2) for d in linspace(0, road_width, num=num_lines)]
                 side_line_offset = 1.5 * self.add_roads_to_scenario.line_width
                 left_side_line = Road('line_white', rid=road.rid + "_left_line")
-                left_side_line_nodes = _calculate_parallel_coords(
-                    initial_line_offsets[0] + side_line_offset, self.add_roads_to_scenario.line_width)
+                offset_list_line_left = _calculate_offset_list(relative_offset=-0.5,
```

Comment:
As I coud see on your screenshots the side lines are right on the edge of the road. For striving for perfection 😉 you could move the side lines a little more into the road just like real roads (https://i.ytimg.com/vi/mGvuSl-HRaY/maxresdefault.jpg).